src/overlays/actors/ovl_En_Zos/z_en_zos.c

```diff
@@ -118,23 +133,26 @@ void EnZos_Destroy(Actor* thisx, PlayState* play) {
     gSaveContext.save.weekEventReg[52] &= (u8)~0x10;
 }
 
-void func_80BBAE84(EnZos* this, s16 arg1, u8 arg2) {
+void EnZos_ChangeAnim(EnZos* this, s16 animIndex, u8 animMode) {
     static AnimationHeader* sAnimations[] = {
-        &object_zos_Anim_00A164, &object_zos_Anim_01621C, &object_zos_Anim_016980, &object_zos_Anim_007C40,
-        &object_zos_Anim_008630, &object_zos_Anim_0090CC, &object_zos_Anim_009830, &object_zos_Anim_005E04,
-        &object_zos_Anim_007334, &object_zos_Anim_0048D4, &object_zos_Anim_001CFC, &object_zos_Anim_005E04,
-        &object_zos_Anim_007334,
+        &gEvanLeanOnKeyboardAnim,  &gEvanLeanOnKeyboardAndSighAnim,
+        &gEvanHandsOnHipsAnim,     &gEeanTalkFootTapAnim,
+        &gEvanTalkLookDownAnim,    &gEvanTalkArmsOutAnim,
+        &gEvanTalkHandsOnHipsAnim, &gEvanPlayRightAnim,
+        &gEvanPlayLeftAnim,        &gEvanInspiredAnim,
+        &gEvanSlowPlayAnim,        &gEvanPlayRightAnim,
+        &gEvanPlayLeftAnim,
     };
-    f32 phi_f0;
+    f32 endFrame;
 
-    if ((arg1 != this->unk_2B8) && (arg1 >= 0) && (arg1 < 13)) {
-        if (arg1 > 10) {
-            phi_f0 = 29.0f;
+    if ((animIndex != this->animIndex) && (animIndex >= 0) && (animIndex < 13)) {
```

Comment:
Make a MAX for the enum and use it here. Maybe put the enums by the animations too, although in this case it's probably not needed.

Reply:
I can't put the enum by the animations, since I need it in `EnZos_Init`. I did everything else though.

Reply:
Ah, I meant in the sense of comments.